tests/views_integration/Tribe/Events/Views/V2/TemplateBootstrapTest.php

```diff
@@ -2,5 +2,186 @@
 namespace Tribe\Events\Views\V2;
 
 class TemplateBootstrapTest extends \Codeception\TestCase\WPTestCase {
+	public function setUp() {
+		// before
+		parent::setUp();
+	}
+
+	public function tearDown() {
+		// your tear down methods here
+
+		// then
+		parent::tearDown();
+	}
+
+	private function make_instance() {
+		return new Template_Bootstrap();
+	}
+
+	/**
+	 * @test
+	 * it should be instantiatable
+	 */
+	public function it_should_be_instantiatable() {
+		$sut = $this->make_instance();
+
+		$this->assertInstanceOf( Template_Bootstrap::class, $sut );
+	}
+
+	public function base_template_options() {
+		return [
+			'invalid' => [
+				'foo',
+				'event',
+			],
+			'numeric' => [
+				2,
+				'event',
+			],
+			'default' => [
+				'default',
+				'page',
+			],
+			'empty' => [
+				'',
+				'event',
+			],
+			'null' => [
+				null,
+				'event',
+			],
+			'boolean_false' => [
+				false,
+				'event',
+			],
+			'boolean_true' => [
+				false,
+				'event',
+			],
+			'event' => [
+				'event',
+				'event',
+			],
+			'page_isnt_a_value' => [
+				'page',
+				'event',
+			],
+		];
+	}
+
+	/**
+	 * It should only allow permitted values on base template option
+	 *
+	 * @test
+	 * @dataProvider base_template_options
+	 */
+	public function should_only_allow_permitted_values_on_base_template_option( $input, $expected ) {
+		tribe_update_option( 'tribeEventsTemplate', $input );
+		$option_value = $this->make_instance()->get_template_setting();
+
+		$this->assertEquals( $option_value, $expected );
+	}
+
+	/**
+	 * It should return template event instance
+	 *
+	 * @test
+	 */
+	public function it_should_return_template_event_instance() {
+		tribe_update_option( 'tribeEventsTemplate', 'event' );
+		$instance = $this->make_instance()->get_template_object();
+
+		$this->assertInstanceOf( Template\Event::class, $instance );
+	}
+
+	/**
+	 * It should return template page instance
+	 *
+	 * @test
+	 */
+	public function it_should_return_template_page_instance() {
+		tribe_update_option( 'tribeEventsTemplate', 'default' );
+		$instance = $this->make_instance()->get_template_object();
+
+		$this->assertInstanceOf( Template\Page::class, $instance );
+	}
+
+	public function query_args_to_load() {
+		return [
+			'invalid' => [
+				'foo',
+				false,
+			],
+			'post_type_eq_tribe_events' => [
+				[
+					'post_type' => 'tribe_events',
+				],
+				true,
+			],
+			'post_type_contains_tribe_events' => [
+				[
+					'post_type' => [ 'tribe_events', 'foo' ],
```

Comment:
Split this to have `tribe_events` with a valid test case (e.g. `post`) and an invalid one (e.g. `invalid_post_type`).
If we're not checking if the post types are valid let's make it explicit and clear.